certbot-nginx/certbot_nginx/parser.py

```diff
@@ -82,21 +82,28 @@ def abs_path(self, path):
         else:
             return path
 
-    def get_vhosts(self):
-        # pylint: disable=cell-var-from-loop
-        """Gets list of all 'virtual hosts' found in Nginx configuration.
-        Technically this is a misnomer because Nginx does not have virtual
-        hosts, it has 'server blocks'.
-
-        :returns: List of :class:`~certbot_nginx.obj.VirtualHost`
-            objects found in configuration
-        :rtype: list
+    def _build_addr_to_ssl(self):
+        """Builds a map from address to whether it listens on ssl in any server block
+        """
+        servers = self._get_raw_servers()
 
+        addr_to_ssl = {}
+        for filename in servers:
+            for server, _ in servers[filename]:
+                # Parse the server block to save addr info
+                parsed_server = _parse_server_raw(server)
+                for addr in parsed_server['addrs']:
+                    addr_tuple = addr.normalized_tuple()
+                    if addr_tuple not in addr_to_ssl:
+                        addr_to_ssl[addr_tuple] = addr.ssl
+                    addr_to_ssl[addr_tuple] = addr.ssl or addr_to_ssl[addr_tuple]
```

Comment:
the pattern I'm going for is lazily-create-then-or-in-the-new-value. I can make that more clear by reordering the arguments to `or` and initializing with `False`.